src/Extension/CoreExtension.php

```diff
@@ -420,18 +420,18 @@ public static function random(Environment $env, $values = null, $max = null)
      *
      * @internal
      */
-    public static function dateFormatFilter(Environment $env, $date, $format = null, $timezone = null)
+    public function dateFormatFilter($date, $format = null, $timezone = null)
     {
         if (null === $format) {
-            $formats = $env->getExtension(self::class)->getDateFormat();
+            $formats = $this->getDateFormat();
```

Comment:
The main idea is to remove the usage of `$env` when possible so that we can extract these methods to another class that would be available for public consumption.